### ISSUE TITLE
Make DisplayNameAttribute available in .NET Core

### ISSUE THREAD

**Issue by Joe4evr** [dotnet/corefx#1053]
This was originally going to be a request to move `DisplayNameAttribute` into Core, but the problem seems a little bigger than that. Currently, `System.dll` houses approximately a couple hundred types spread across four dozen different namespaces, but not everything in it is listed on the [Open Source Progress table](https://github.com/dotnet/corefx-progress/blob/master/src-diff/README.md) (although a select few namespaces like `System.Security.Cryptography.X509Certificates` *are* there).

This still makes some namespaces (for example `System.ComponentModel`) feel incomplete because of how they're spread across different assemblies. The previously mentioned `X509Certificates` namespace is already being tracked with all its types from `System.dll`, so other namespaces should follow suit, right?

I was porting a project of mine over to see the ASP.NET vNext templates in action, but the lack of `DisplayNameAttribute` is literally the only thing preventing me from building it (even if I tell it to not target .NET Core) without commenting those attributes out. And if .NET Core's philosophy is to only use what you need, why should I be forced to import some monolithic DLL with hundreds of types that I don't need just for the one that I *do* need, because that type is not in the DLL named after its containing namespace? ¯\\(°_o)/¯

**Comment by ellismg** [comment]
The entries in the progress table on that webpage are not namespaces per say, but rather the names of assemblies.  As part of the .NET Core effort, we've refactored the monolithic assemblies like System.dll into a set of smaller contracts, which make up .NET Core.  This factoring was done not on namespace boundaries but by grouping related types that form cohesive groups and version at a similar cadence together.

The goal of .NET Core was not to provide every type which was in System.dll, mscorlib.dll, etc into a set of contracts.  Rather, we looked at the set of types which made sense for the foundational libraries for .NET across different devices and runtimes (e.g. the Full .NET Framework 4.6, ASP.NET vNext on top of CoreCLR and .NET Native).

As to the specific reason that DisplayNameAttribute wasn't included @davkean or @terrajobst probably have more context.  It's possible that there is a different type that serves a similar purpose that could be used instead.

We're excited to get community feedback on what types we're missing that are used in common scenarios, so keep feedback like the above coming.  I just wanted to take a moment to call out that factoring the entire .NET Framework for .NET Core was an explicit non goal.
